internal/job/hook/binary.go

```diff
@@ -60,6 +60,16 @@ func isBinaryExecutable(path string) (bool, error) {
 		return false, fmt.Errorf("reading first four bytes of file %q: %w", path, err)
 	}
 
+	fileInfo, err := f.Stat()
+	if err != nil {
+		return false, fmt.Errorf("stat file %q: %w", path, err)
+	}
+
+	if fileInfo.Size() < 4 {
+		// there are less than four bytes in the file, there's nothing that we can do with it
+		return false, nil
+	}
```

Comment:
I think we should move this in between line 56 `defer f.Close()` and line 57 `r := bufio.NewReader(f)`, so that it checks the size before trying to `r.Peek(4)`.